src/EditorFeatures/Core.Wpf/Structure/AbstractStructureTaggerProvider.cs

```diff
@@ -94,6 +94,13 @@ protected sealed override ITaggerEventSource CreateEventSource(ITextView textVie
         {
             try
             {
+                // Let LSP handle producing tags in the cloud scenario
+                var workspaceContextService = documentSnapshotSpan.Document.Project.Solution.Workspace.Services.GetService<IWorkspaceContextService>();
```

Comment:
this can be getrequiredservice

and potentially we could just make an extension on document and just use that for all of these
http://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Shared/Extensions/ITextBufferExtensions.cs,51